src/build_helper/run.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2012-2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Run the command in a background thread. Redirect the output (and error
+//! output) to the screen.
+
+use std::process::{Stdio, Command};
+use std::sync::mpsc::{channel, Receiver};
+use std::thread;
+use std::io::{self, Read, BufRead, BufReader, Write};
+
+/// Run the command in another thread. Print output to screen.
+/// Panics if command failed to run.
+pub trait Run {
+    fn run(&mut self);
+}
+
+#[derive(Clone, Copy)]
+enum OutputType { Stdout, Stderr }
+
+impl Run for Command {
+    fn run(&mut self) {
+        match run(self) {
+            Ok(_) => {},
+            Err(msg) => panic!(msg)
+        }
+    }
+}
+
+fn run(cmd : &mut Command) -> Result<(), String> {
+    println!("Running {:?}", cmd);
+    let mut child = try!(cmd.stdin(Stdio::piped())
+                         .stdout(Stdio::piped())
+                         .stderr(Stdio::piped())
```

Comment:
How come this creates a pipe if we just turn around and redirect them back to stdout?